staging/src/k8s.io/code-generator/cmd/applyconfiguration-gen/generators/packages.go

```diff
@@ -236,8 +243,10 @@ func packageTypesForInputDirs(context *generator.Context, inputDirs []string, ou
 			klog.Warningf("Skipping internal package: %s", p.Path)
 			continue
 		}
-		gv := groupVersion(p)
-		pkg := filepath.Join(outputPath, gv.Group.PackageName(), strings.ToLower(gv.Version.NonEmpty()))
+		// this is how the client generator finds the package we are creating. It uses the API package name, not the group name.
+		_, gvPackageString := util.ParsePathGroupVersion(p.Path)
+
+		pkg := filepath.Join(outputPath, strings.ToLower(gvPackageString))
```

Comment:
does gvPackageString do the right thing with transforming `api` → `core`, etc? without test data, I'm not sure what this change is accomplishing

Reply:
I tried to explain the comment above.  I'll clarify there. 

> It uses the API package name, not the group name.

so for openshift/api/cloudnetwork/v1, if the group is cloud.network.openshift.io, the client generator will choose cloudnetwork as the package for the generated client, but before this change the apply generator would choose cloud.  This change makes the applygenerator match.

This would be breaking, but since the generated client doesn't work if these don't match, I don't think anyone has used it.